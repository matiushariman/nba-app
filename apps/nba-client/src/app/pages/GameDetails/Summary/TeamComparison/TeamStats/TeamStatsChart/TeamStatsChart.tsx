import type { TeamStatsChartProps } from './TeamStatsChart.types';

const COLORS = {
  home: 'bg-red-800',
  away: 'bg-blue-800',
};

const TeamStatsChart = ({
  teamTricode,
  percentage = 0,
  secondaryText,
  primaryText,
  color = 'home',
}: TeamStatsChartProps) => {
  return (
    <div>
      <div className="flex items-center gap-2">
        <div className="flex-grow">
          <p className="text-sm">{teamTricode}</p>
        </div>
        {secondaryText && (
          <div>
            <p className="text-gray-500 text-sm">{secondaryText}</p>
          </div>
        )}
        <div>
          <p className="font-bold text-lg">{primaryText}</p>
        </div>
      </div>
      <div className="relative h-4">
        <div
          className={`absolute h-4 ${COLORS[color]} z-10`}
          style={{ width: `${percentage}%` }}
        />
        <div className="absolute h-4 bg-gray-100 w-full" />
      </div>
    </div>
  );
};

export default TeamStatsChart;
