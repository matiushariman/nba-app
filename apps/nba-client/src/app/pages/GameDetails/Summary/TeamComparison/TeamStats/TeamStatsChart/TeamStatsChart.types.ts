import type { Team } from '@nba-app/types-nba';

export interface TeamStatsChartProps extends Pick<Team, 'teamTricode'> {
  readonly percentage?: number;
  readonly secondaryText?: string;
  readonly primaryText: string;
  readonly color?: 'home' | 'away';
}
