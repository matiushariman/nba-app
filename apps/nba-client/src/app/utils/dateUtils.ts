import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import timezone from 'dayjs/plugin/timezone';

dayjs.extend(utc);
dayjs.extend(timezone);

interface ConvertDateToEtParams {
  date?: string | number | dayjs.Dayjs | Date;
  format?: string;
}

export const convertDateToEt = ({
  date,
  format = 'MMMM D, YYYY h:mm A',
}: ConvertDateToEtParams) => dayjs(date).tz('America/Toronto').format(format);
