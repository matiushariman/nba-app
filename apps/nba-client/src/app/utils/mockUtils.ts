export const shouldUseMock = process.env['NODE_ENV'] === 'development';
