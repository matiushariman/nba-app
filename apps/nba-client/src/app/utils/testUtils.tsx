import { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';

import '../../styles.css';

const queryClient = new QueryClient();

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => (
  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';

export { customRender as render };
