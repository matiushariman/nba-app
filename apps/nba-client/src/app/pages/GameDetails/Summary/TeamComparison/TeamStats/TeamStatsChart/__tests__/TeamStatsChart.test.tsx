import TeamStatsChart from '..';

import { render } from '../../../../../../../utils/testUtils';

describe('pages/GameDetails/Summary/TeamComparison/TeamStats/TeamStatsChart', () => {
  it('should render TeamStatsChart that matches snapshot', () => {
    const { baseElement } = render(
      <TeamStatsChart
        primaryText="100/100"
        teamTricode="BOS"
        secondaryText="100%"
      />
    );

    expect(baseElement).toMatchSnapshot();
  });
});
