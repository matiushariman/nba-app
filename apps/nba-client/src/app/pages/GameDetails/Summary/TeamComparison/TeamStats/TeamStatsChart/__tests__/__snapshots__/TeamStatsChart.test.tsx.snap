// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/GameDetails/Summary/TeamComparison/TeamStats/TeamStatsChart should render TeamStatsChart that matches snapshot 1`] = `
<body>
  <div>
    <div>
      <div
        class="flex items-center gap-2"
      >
        <div
          class="flex-grow"
        >
          <p
            class="text-sm"
          >
            BOS
          </p>
        </div>
        <div>
          <p
            class="text-gray-500 text-sm"
          >
            100%
          </p>
        </div>
        <div>
          <p
            class="font-bold text-lg"
          >
            100/100
          </p>
        </div>
      </div>
      <div
        class="relative h-4"
      >
        <div
          class="absolute h-4 bg-red-800 z-10"
          style="width: 0%;"
        />
        <div
          class="absolute h-4 bg-gray-100 w-full"
        />
      </div>
    </div>
  </div>
</body>
`;
